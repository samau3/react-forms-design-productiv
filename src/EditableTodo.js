import React, { useState } from "react";
import Todo from "./Todo";
import TodoForm from "./TodoForm";

/** Show editable todo item.
 *
 * Props
 * - todo
 * - update(): fn to call to update a todo
 * - remove(): fn to call to remove a todo
 *
 * EditableTodoList -> EditableTodo -> { Todo, TodoForm }
 */

function EditableTodo({ todo, update, remove }) {
  console.log("EditableTodo:", todo);
  const { id, title, description, priority } = todo;
  // Is there another way to do this without a state?
  const [isEditing, setIsEditing] = useState(false);
  console.log("isEditing:", isEditing);

  /** Toggle if this is being edited */
  function toggleEdit() { 
    setIsEditing(curr => !curr);
  }
  
  /** Call remove fn passed to this. */
  function handleDelete() { 
    remove(id);
  }
  
  /** Edit form saved; toggle isEditing and update in ancestor. */
  function handleSave() { 
    update(todo);
    toggleEdit();
  }

  return (
      <div className="EditableTodo">
                {isEditing && 
                <TodoForm initialFormData={todo} handleSave={handleSave}/>
                }

                <div className="mb-3">
                  <div className="float-right text-sm-right">
                    <button
                        className="EditableTodo-toggle btn-link btn btn-sm"
                        onClick={toggleEdit}>
                      Edit
                    </button>
                    <button
                        className="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                        onClick={handleDelete}>
                      Del
                    </button>
                  </div>
                  <Todo 
                    id={id} 
                    title={title} 
                    description={description} 
                    priority={priority} 
                  />
                </div>

      </div>
  );
}

export default EditableTodo;
