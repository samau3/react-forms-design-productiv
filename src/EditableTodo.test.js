import React from "react";
import { render } from "@testing-library/react";

import TEST_DATA from "./_testCommon";
import EditableTodo from "./EditableTodo";

it("renders without crashing", function () {
    render(<EditableTodo todo={TEST_DATA[0]} />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<EditableTodo todo={TEST_DATA[0]} />);
    expect(asFragment()).toMatchSnapshot();
});