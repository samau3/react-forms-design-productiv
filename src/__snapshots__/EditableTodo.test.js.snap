// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="EditableTodo"
  >
    <div
      class="mb-3"
    >
      <div
        class="float-right text-sm-right"
      >
        <div
          class="Todo"
        >
          <div>
            <b>
              Code!
            </b>
             
            <small>
              (priority: 2)
            </small>
          </div>
          <div>
            <small>
              Write some code
            </small>
          </div>
        </div>
        <button
          class="EditableTodo-toggle btn-link btn btn-sm"
        >
          Edit
        </button>
        <button
          class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
        >
          Del
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
